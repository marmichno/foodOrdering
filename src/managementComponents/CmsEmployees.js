import {CmsNavbar} from './CmsNavbar';
import {useEffect, useState} from 'react';
import {employeesGetRequest} from '../requests/employeesGetRequest';
import {employeesPostRequest} from '../requests/employeesPostRequest';

export const CmsEmployees = () => {

    const [employeeFirstName, setEmployeeFirstName] = useState("");
    const [employeeLastName, setEmployeeLastName] = useState("");
    const [employeePhoneNumber, setEmployeePhoneNumber] = useState("");
    const [employeeRole, setEmployeeRole] = useState("");
    const [employeeLogin, setEmployeeLogin] = useState("");
    const [employeePasword, setEmployeePasword] = useState("");

    return(
        <div className="cmsEmployeesContainer">
            <CmsNavbar/>
            {/* <div className="cmsEmployeesContent">
                <div className="employeeRow">
                    <div><h1>Id</h1></div>
                    <div><h1>First name</h1></div>
                    <div><h1>Last name</h1></div>
                    <div><h1>Phone number</h1></div>
                    <div><h1>Role</h1></div>
                    <div><h1>Login</h1></div>
                    <div><h1>Password</h1></div>
                    <div><h1>Modify</h1></div>
                </div>

                <div className="employeeRow">
                    <div><p>-</p></div>
                    <div onChange={(e) => setEmployeeFirstName(e.target.value)} className="addInput"><input type="text" placeholder="Marcin" autocomplete="off" name="category" required></input></div>
                    <div onChange={(e) => setEmployeeLastName(e.target.value)} className="addInput"><input type="text" placeholder="Dyl" autocomplete="off" name="category" required></input></div>
                    <div onChange={(e) => setEmployeePhoneNumber(e.target.value)} className="addInput"><input type="text" placeholder="123-245-638" autocomplete="off" name="category"></input></div>
                    <div className="addInput">
                        <select>
                            <option>Sushi Chef</option><option>Chef</option><option>Waiter</option><option>Manager</option>
                        </select>
                    </div>
                    <div onChange={(e) => setEmployeeLogin(e.target.value)} className="addInput"><input type="text" placeholder="Login" autocomplete="off" name="category" required></input></div>
                    <div onChange={(e) => setEmployeePassword(e.target.value)} className="addInput"><input type="password" placeholder="password" autocomplete="off" name="category" required></input></div>
                    <div><button>Add</button></div>
                </div>

                {employees.map(value => {
                   return(
                    <div className="employeeRow">
                        <div><p>9</p></div>
                        <div><p>Marcin</p></div>
                        <div><p>Dyl</p></div>
                        <div><p>123-234-234</p></div>
                        <div><p>Chef</p></div>
                        <div><p>mdyl</p></div>
                        <div><p>-</p></div>
                        <div><button>Modify</button><button>Delete</button></div>
                    </div>
                   ) 
                })}
            </div> */}
        </div>
    )
}