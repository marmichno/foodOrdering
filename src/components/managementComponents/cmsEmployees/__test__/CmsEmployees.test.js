jest.unmock('axios');
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {CmsEmployees} from '../CmsEmployees';
import "@testing-library/jest-dom/extend-expect";
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux'; 
import validEmployeesJson from '../__mocks__/validEmployees.json';
import validEmployeesRolesJson from '../__mocks__/validEmployeesRoles.json';
import TestRenderer from "react-test-renderer";

const store = createStore(() => [], {}, applyMiddleware());

const mock = new MockAdapter(axios);

test("checks if employees are visible after api call", async () => {

    mock.onGet("http://localhost:8080/api/employee").reply(200, validEmployeesJson);
    mock.onGet("http://localhost:8080/api/role").reply(200, validEmployeesRolesJson);

    const {getAllByText} = render(<Provider store={store}><CmsEmployees/></Provider>);

    await waitFor(() => {
        expect(getAllByText('Dyl'));
        expect(getAllByText('123234123'));
        expect(getAllByText('sushi chef'));
        expect(getAllByText('asd'));
    })
});

test("checks if employees roles are visible after api call", async () => {

    mock.onGet("http://localhost:8080/api/role").reply(200, validEmployeesRolesJson);
    mock.onGet("http://localhost:8080/api/employee").reply(200, validEmployeesJson);

    const {getAllByText} = render(<Provider store={store}><CmsEmployees/></Provider>);

    await waitFor(() => {
        expect(getAllByText('sushi chef'));
    })
});

test("matches snapshot", () => {
    const tree = TestRenderer.create(<Provider store={store}><CmsEmployees/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});
