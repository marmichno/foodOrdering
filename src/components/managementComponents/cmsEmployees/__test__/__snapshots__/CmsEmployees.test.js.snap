// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cmsEmployeesContainer"
>
  <div
    className="hamburgerMenu"
  >
    <svg
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="cmsEmployeesContent"
  >
    <div
      className="employeeRow"
    />
    <div
      className="employeeRow"
    >
      <div>
        <h1>
          Id
        </h1>
      </div>
      <div>
        <h1>
          First name
        </h1>
      </div>
      <div>
        <h1>
          Last name
        </h1>
      </div>
      <div>
        <h1>
          Phone number
        </h1>
      </div>
      <div>
        <h1>
          Role
        </h1>
      </div>
      <div>
        <h1>
          Login
        </h1>
      </div>
      <div>
        <h1>
          Password
        </h1>
      </div>
      <div>
        <h1>
          Modify
        </h1>
      </div>
    </div>
    <div
      className="employeeRow"
    >
      <div>
        <p>
          -
        </p>
      </div>
      <div
        className="addInput"
        onChange={[Function]}
      >
        <input
          autoComplete="off"
          name="category"
          placeholder="Marcin"
          required={true}
          type="text"
        />
      </div>
      <div
        className="addInput"
        onChange={[Function]}
      >
        <input
          autoComplete="off"
          name="category"
          placeholder="Dyl"
          required={true}
          type="text"
        />
      </div>
      <div
        className="addInput"
        onChange={[Function]}
      >
        <input
          autoComplete="off"
          name="category"
          placeholder="123-245-638"
          type="text"
        />
      </div>
      <div
        className="addInput"
      >
        <select
          onChange={[Function]}
        />
      </div>
      <div
        className="addInput"
        onChange={[Function]}
      >
        <input
          autoComplete="off"
          name="category"
          placeholder="Login"
          required={true}
          type="text"
        />
      </div>
      <div
        className="addInput"
        onChange={[Function]}
      >
        <input
          autoComplete="off"
          name="category"
          placeholder="password"
          required={true}
          type="password"
        />
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
